inginious/frontend/plugins/upcoming_tasks/__init__.py

```diff
@@ -0,0 +1,142 @@
+        if string_time_planner in ["7", "14", "30"]:
+            return int(string_time_planner)
+        return 100000
+
+    def page(self, time_planner):
+        """ General main method called for GET and POST """
+        username = self.user_manager.session_username()
+        all_courses = self.course_factory.get_all_courses()
+        time_planner = self.time_planner_conversion(time_planner)
+
+        # Get the courses id
+        open_courses = {courseid: course for courseid, course in all_courses.items()
+                        if self.user_manager.course_is_open_to_user(course, username, False) and
+                        self.user_manager.course_is_user_registered(course, username)}
+        open_courses = OrderedDict(sorted(iter(open_courses.items()), key=lambda x: x[1].get_name(self.user_manager.session_language())))
+
+        # Get last submissions for left panel
+        last_submissions = self.submission_manager.get_user_last_submissions(5, {"courseid": {"$in": list(open_courses.keys())}})
+        except_free_last_submissions = []
+        for submission in last_submissions:
+            try:
+                submission["task"] = open_courses[submission['courseid']].get_task(submission['taskid'])
+                except_free_last_submissions.append(submission)
+            except:
+                pass
+
+        # Get the courses tasks, remove finished ones and courses that have no available unfinished tasks with upcoming deadline in range
+        tasks_data = {}
+        succeeded_courses = []
+        for courseid, course in open_courses.items():
+            tasks_data[courseid] = {}
+            tasks = course.get_tasks()
+            outdated_tasks = [taskid for taskid, task in tasks.items() if (not task.get_accessible_time().is_open()) or ((task.get_accessible_time().get_soft_end_date()) > (datetime.now()+timedelta(days=time_planner)))]
+            new_user_task_list = course.get_task_dispenser().get_user_task_list([username])[username]
+            new_user_task_list = [task_id for task_id in new_user_task_list if task_id not in outdated_tasks]
+            tasks_data[courseid] = {taskid: {"succeeded": False, "grade": 0.0} for taskid in new_user_task_list}
+            user_tasks = self.database.user_tasks.find({"username": username, "courseid": course.get_id(), "taskid": {"$in": new_user_task_list}})
+            for user_task in user_tasks:
+                if not user_task["succeeded"]:
+                    tasks_data[courseid][user_task["taskid"]]["succeeded"] = user_task["succeeded"]
+                    tasks_data[courseid][user_task["taskid"]]["grade"] = user_task["grade"]
+                else:
+                    del tasks_data[courseid][user_task["taskid"]]
+            # Remove courses with no unfinished available tasks with deadline and lti courses
+            if (not any(task in tasks for task in tasks_data[courseid])) or course.is_lti():
```

Comment:
any(task in tasks for task in tasks_data[courseid])
    
   Are you actually checking for a non-zero length ?